app/components/widget-math.js

```diff
@@ -0,0 +1,79 @@
+import Component from '@ember/component';
+import WidgetScriptStateMixin from '../mixins/widget-script-state';
+
+/**
+ * MathJax
+ * @typedef {Object} MathJax
+ */
+
+/**
+ * Window
+ * @typedef {Object} Window
+ * @property {MathJax} [MathJax]
+ */
+
+export default Component.extend(
+	WidgetScriptStateMixin,
+	{
+		tagName: 'span',
+		classNames: ['widget-math'],
+		data: null,
+
+		/**
+		 * @returns {void}
+		 */
+		didInsertElement() {
+			this._super(...arguments);
+
+			this.loadScript();
+			this.renderMath();
+		},
+
+		/**
+		 * @returns {void}
+		 */
+		loadScript() {
+			if (!this.get('scriptLoadInitialized.mathjax')) {
+				// initialize the loading of MathJax just once
+				this.set('scriptLoadInitialized.mathjax', true);
+
+				// let's inject a node with MathJax config
+				document.head.insertAdjacentHTML('beforeend', `
+	<script type="text/x-mathjax-config">
+	MathJax.Hub.Config({
+		extensions: ["tex2jax.js"],
+		jax: ["input/TeX", "output/HTML-CSS"],
+		// we will render only specific nodes
+		skipStartupTypeset: true,
+		tex2jax: {
+			inlineMath: [ ['$','$'], ["\\(","\\)"] ],
+			displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
+			processEscapes: true
+		},
+		"HTML-CSS": { fonts: ["TeX"] }
+	});
+
+	// cast NodeList to an array
+	var elements = [].slice.call(window.document.body.querySelectorAll('.tex'));
+
+	window.MathJax.Hub.Queue(["Typeset", MathJax.Hub, elements]);
```

Comment:
I think these two lines are not needed.

Reply:
Well, they are. `renderMath` method only does rendering when the MathJax library is loaded. Hence, the first renderings need to happen via this code.

Reply:
But isn't this code also run when the MatJax library is loaded? That's why it's put in `<script type="text/x-mathjax-config">` which isn't parsed by browser. I think we do the same thing twice, here and in `renderMath`

Reply:
`this.renderMath` is required to re-render the math tags (when you revisit the page with tags) or when you hit the page with tags and the MathJax is already loaded (i.e. was loaded as a part of a previous article view).

Reply:
Ok, I went through it again and now I see that the first time `renderMath` is called it does nothing.